### PR TITLE
Reactive Armors

### PR DESCRIPTION
**Pull request by SmArtKar**
## About The Pull Request

Now each anomaly gets it's own type of armor! Pyro gets a fire one with a reworked effect, while vortex gets stealth

## Why It's Good For The Game

Finally pyro reactive armor makes sense

## Changelog
:cl:
tweak: Pyro Reactive Armor now creates a ball of fire around the user while extinguishing them! Nice!
/:cl:

### CODE REVIEW

File: code/modules/clothing/suits/reactive_armour.dm [hunk 1]
```diff
@@ -112,11 +114,16 @@
...
+		playsound(get_turf(src), 'sound/items/welder.ogg', 75, TRUE)
+		var/_range = 1
+		for(var/i = 0, i <= 2,i++)
+			for(var/turf/T in spiral_range_turfs(_range,get_turf(src)))
+				new /obj/effect/hotspot(T)
+				T.hotspot_expose(700,50,1)
+				for(var/mob/living/livies in T.contents - get_turf(src))
+					livies.adjustFireLoss(5)
+			_range++
+			sleep(3)
```
Comment (XDTM) [inline code/modules/clothing/suits/reactive_armour.dm:126]: I'm fairly sure that sleeps are to be avoided whenever possible according to code standards, although the only other solution i can think of is using a recursive timer call.

**Comment by Pugie** [comment]
Cool idea, don't think it should exist without a cooldown. I think the same thing about every reactive armour though otherwise they're way too strong.

File: code/modules/clothing/suits/reactive_armour.dm (hunk 1 above)
Comment (SmArtKar) [reply code/modules/clothing/suits/reactive_armour.dm:126]: sleeps are here for better animation, and no, everybody puts sleeps

**Comment by SmArtKar** [comment]
> Cool idea, don't think it should exist without a cooldown. I think the same thing about every reactive armour though otherwise they're way too strong.

It has a cooldown. A pretty big one actually

**Comment by Pugie** [comment]
Really? I was just testing it out in local by punching a guy with reactive fire armour and every single punch I threw triggered the armour? I'm pretty sure the only armour with an actual long cooldown is teleport.

**Comment by XDTM** [comment]
`reactivearmor_cooldown_duration` seems to be set to 0 on all armors except the teleport armor, which has a 10 seconds cooldown.

**Comment by XDTM** [comment]
`reactivearmor_cooldown_duration` seems to be set to 0 on all armors except the teleport armor, which has a 10 seconds cooldown.

**Comment by Winterous** [comment]
Wait, what? Vortex already did stealth.

**Comment by moocowswag** [comment]
> Wait, what? Vortex already did stealth.

Both pyro armor and vortex got stealth, now pyro is getting its own and vortex will continue to be for stealth armor.

**Comment by ATH1909** [comment]
I tried PRing the fire armor back into the game, but the PR got denied because it's apparently cancerous in places like the emergency shuttle, especially when being worn by a non-antag.

**Comment by ATH1909** [comment]
See: https://github.com/tgstation/tgstation/pull/51274.

**Comment by SmArtKar** [comment]
> > Wait, what? Vortex already did stealth.
> 
> Both pyro armor and vortex got stealth, now pyro is getting its own and vortex will continue to be for stealth armor.
> 
> I just dont like how situationally strong / weak this armor might be.
> 
> I dont know how big the fire aoe is but this will probably be the only reactive armor that does nothing against someone who is hitting you from range OR wearing a hardsuit. All of the other armors almost always have some effect aside from the block chance. (Tesla armor for example is only really countered by being a borg/having insulated mutation/wearing CE hardsuit)
> 
> And reactive tesla armor already stomps on crowds of greytiders, although this will probably take away their clothes in the process.
> 
> Maybe instead of making fire and setting everyone in the aoe on fire it heats everyone up, more than enough to take damage (even through their hardsuits) and maybe let it be the reactive armor variant that has the lowest cooldown (if all of them eventually get one, maybe even let it keep a cooldown of zero)

area is 5x5

**Comment by Pugie** [comment]
> I just dont like how situationally strong / weak this armor might be.

It's still 50% block versus mostly any attack it being situational shouldn't be a worry.
Table armour is a viable choice which stands testament to that.

File: code/modules/clothing/suits/reactive_armour.dm [hunk 2]
```diff
@@ -10,7 +10,9 @@
 	var/static/list/anomaly_armour_types = list(
 		/obj/effect/anomaly/grav	                = /obj/item/clothing/suit/armor/reactive/repulse,
 		/obj/effect/anomaly/flux 	           		= /obj/item/clothing/suit/armor/reactive/tesla,
-		/obj/effect/anomaly/bluespace 	            = /obj/item/clothing/suit/armor/reactive/teleport
+		/obj/effect/anomaly/bluespace 	            = /obj/item/clothing/suit/armor/reactive/teleport,
+		/obj/effect/anomaly/pyro					= /obj/item/clothing/suit/armor/reactive/fire,
+		/obj/effect/anomaly/bhole					= /obj/item/clothing/suit/armor/reactive/stealth
```
Comment (Jared-Fogle) [inline code/modules/clothing/suits/reactive_armour.dm:15]: Can you remove the tabs before the `=` and just have a space? It's hideous.

**Pull request closed without merging**